app/scripts/lib/ppom/ppom-middleware.ts

```diff
@@ -78,9 +81,13 @@ export function createPPOMMiddleware<
       const securityAlertsEnabled =
         preferencesController.store.getState()?.securityAlertsEnabled;
 
-      const { chainId } = getProviderConfig({
-        metamask: networkController.state,
-      });
+      // This will always exist as the SelectedNetworkMiddleware
+      // adds networkClientId to the request before this middleware runs
+      const { chainId } =
```

Comment:
worth mentioning that `SelectedNetworkController` middleware (and the middleware to tag on the networkClientId in the full multichain API now **has to** come before this middleware)

Reply:
added comment to MetaMaskController here https://github.com/MetaMask/metamask-extension/pull/27263/commits/d30b4a53108a221e71b122a2450ffd413e74cf0e